WIP [Federation][e2e] Retrieve member cluster config from secrets instead of kubeconfig

marun opened the pull request:
Previously the federation e2e setup was reading member cluster configuration from the test run's configured kubeconfig.  This change removes that dependency in favor of reading member cluster configuration from secrets in the hosting cluster.

cc: @kubernetes/sig-federation-pr-reviews

marun committed:
fed: Refactor e2e cluster functions into framework for reuse

marun committed:
fed: Refactor e2e setup to retrieve cluster config from secrets

Previously the federation e2e setup was reading member cluster
configuration from the test run's configured kubeconfig.  This change
removes that dependency in favor of reading member cluster
configuration from secrets in the hosting cluster.

k8s-reviewable commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.kubernetes.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/44006)
<!-- Reviewable:end -->

marun committed:
fed: remove unused e2e framework

marun committed:
fed: Refactor e2e backend pod management for maintainability

This change separates backend pod management from cluster management

marun committed:
fed: Manage e2e clusters with a slice instead of a map

There was no code relying on a map, and using a slice eliminates the
need to select the primary cluster when retrieving registered
clusters.

marun committed:
fed: Cache e2e cluster configuration
